src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticHelper.cs

```diff
@@ -0,0 +1,147 @@
+            foreach (var invocation in invocationReferences.OrderByDescending(n => n.Span.Start))
+            {
+                var currentInvocation = root.GetCurrentNode(invocation);
+
+                var seenNamedArgument = currentInvocation.ArgumentList.Arguments.Any(a => a.NameColon != null);
+                var seenDefaultArgumentValue = currentInvocation.ArgumentList.Arguments.Count < localFunction.ParameterList.Parameters.Count;
+
+                var newArguments = parameters.Select(
+                    p => CSharpSyntaxGenerator.Instance.Argument(
+                        seenNamedArgument || seenDefaultArgumentValue ? p.Name : null,
+                        p.RefKind,
+                        p.Name.ToIdentifierName()) as ArgumentSyntax);
+
+                var newArgList = currentInvocation.ArgumentList.WithArguments(currentInvocation.ArgumentList.Arguments.AddRange(newArguments));
+                var newInvocation = currentInvocation.WithArgumentList(newArgList);
+
+                root = root.ReplaceNode(currentInvocation, newInvocation);
+            }
+
+            // Updates the local function declaration with variables passed in as parameters
+            localFunction = root.GetCurrentNode(localFunction);
+            var localFunctionWithNewParameters = CodeGenerator.AddParameterDeclarations(
+                localFunction,
+                parameters,
+                document.Project.Solution.Workspace);
+
+            if (shouldWarn)
+            {
+                var annotation = WarningAnnotation.Create(CSharpFeaturesResources.Warning_colon_Adding_parameters_to_local_function_declaration_may_produce_invalid_code);
+                localFunctionWithNewParameters = localFunctionWithNewParameters.WithAdditionalAnnotations(annotation);
+            }
+
+            var fixedLocalFunction = AddStaticModifier(localFunctionWithNewParameters, CSharpSyntaxGenerator.Instance);
+            var fixedRoot = root.ReplaceNode(localFunction, fixedLocalFunction);
+
+            return document.WithSyntaxRoot(fixedRoot);
+        }
+
+        public static SyntaxNode AddStaticModifier(SyntaxNode localFunction, SyntaxGenerator generator)
+            => generator.WithModifiers(
+                localFunction,
+                generator.GetModifiers(localFunction).WithIsStatic(true));
+
+        /// <summary>
+        /// Creates a new parameter symbol for each captured variables.
+        /// </summary>
+        private static ImmutableArray<IParameterSymbol> CreateParameterSymbols(ImmutableArray<ISymbol> captures)
+        {
+            var parameters = ArrayBuilder<IParameterSymbol>.GetInstance(captures.Length);
+
+            foreach (var symbol in captures)
+            {
+                parameters.Add(CodeGenerationSymbolFactory.CreateParameterSymbol(
+                    attributes: default,
+                    refKind: RefKind.None,
+                    isParams: false,
+                    type: symbol.GetSymbolType(),
```

Comment:
>GetSymbolType [](start = 33, length = 13)

[GetSymbolType](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Shared/Extensions/ISymbolExtensions.cs,875) can return null. Can we assert it is non-null for this callsite?

Reply:
I think captures can only be local, parameter, field or property. So this should be fine?